logging.go

```diff
@@ -1,183 +1,272 @@
 package amqprpc
 
 import (
-	"fmt"
-	"sort"
-	"strings"
+	"log/slog"
 
 	amqp "github.com/rabbitmq/amqp091-go"
 )
 
-/*
-LogFunc is used for logging in amqp-rpc. It makes it possible to define your own logging.
-
-Here is an example where the logger from the log package is used:
-
-	debugLogger := log.New(os.Stdout, "DEBUG - ", log.LstdFlags)
-	errorLogger := log.New(os.Stdout, "ERROR - ", log.LstdFlags)
-
-	server := NewServer(url)
-	server.WithErrorLogger(errorLogger.Printf)
-	server.WithDebugLogger(debugLogger.Printf)
-
-It can also be used with for example a Logrus logger:
-
-	logger := logrus.New()
-	logger.SetLevel(logrus.DebugLevel)
-	logger.Formatter = &logrus.JSONFormatter{}
-
-	s.WithErrorLogger(logger.Warnf)
-	s.WithDebugLogger(logger.Debugf)
-
-	client := NewClient(url)
-	client.WithErrorLogger(logger.Errorf)
-	client.WithDebugLogger(logger.Debugf)
-*/
-type LogFunc func(format string, args ...interface{})
-
-func stringifyTableForLog(v amqp.Table) string {
+// logAttrsFromTable returns a slice of slog.Attr from an amqp.Table. This will
+// NOT filter out sensitive information. That should be done in a custom
+// slog.Handler together with the users own application code.
+func logAttrsFromTable(v amqp.Table) []slog.Attr {
 	if len(v) == 0 {
-		return "[]"
+		return nil
 	}
 
-	vals := []string{}
+	attrs := []slog.Attr{}
 
 	for key, val := range v {
 		if inner, ok := val.(amqp.Table); ok {
-			val = stringifyTableForLog(inner)
-		}
-
-		strVal := fmt.Sprintf("%v", val)
+			attrs = append(attrs, slog.Attr{
+				Key:   key,
+				Value: slog.GroupValue(logAttrsFromTable(inner)...),
+			})
 
-		if strVal == "" {
 			continue
 		}
 
-		vals = append(vals, fmt.Sprintf("%s=%s", key, strVal))
+		attrs = append(attrs, slog.Any(key, val))
 	}
 
-	sort.Strings(vals)
-
-	return fmt.Sprintf("[%s]", strings.Join(vals, ", "))
+	return attrs
 }
 
-func stringifyDeliveryForLog(v *amqp.Delivery) string {
+func logAttrsForDelivery(v *amqp.Delivery) []slog.Attr {
 	if v == nil {
-		return "[nil]"
+		return nil
 	}
 
-	vals := []string{}
-
-	if v.Exchange != "" {
-		vals = append(vals, fmt.Sprintf("Exchange=%s", v.Exchange))
+	vals := []slog.Attr{
+		slog.String("correlation_id", v.CorrelationId),
+		slog.String("exchange", v.Exchange),
+		slog.Bool("redelivered", v.Redelivered),
 	}
 
 	if v.RoutingKey != "" {
-		vals = append(vals, fmt.Sprintf("RoutingKey=%s", v.RoutingKey))
+		vals = append(vals, slog.String("routing_key", v.RoutingKey))
```

Comment:
Do you think it's ugly to log empty strings/nulls? Thinking now with structured logging should we just always add these attributes? I don't see a problem with a log like

```json
{"level":"info",  "content_type": "application/json", "content_encoding": "", "delivery_mode": ""}
```

But I also don't see much issues with all these if statements so just a question 😄

Reply:
Actually I changed my mind. Let's avoid logging empty/non expected values. The logs will just be unnecessary big and noisy!